CRM/Member/BAO/MembershipType.php

```diff
@@ -128,6 +139,20 @@ public static function add(&$params, $ids = array()) {
     if ($id) {
       self::updateAllPriceFieldValue($id, $params);
     }
+
+    if (!empty($params['custom']) &&
```

Comment:
@mattwire this should already be supported in the api layer  in 5.0  see 
https://github.com/civicrm/civicrm-dev-docs/commit/b0bf2bac2917b04ff617be32327be0eda3c80f16

We don't need to add via bao layer